elasticsearch-transport/spec/elasticsearch/connections/selector_spec.rb

```diff
@@ -0,0 +1,159 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+require 'spec_helper'
+
+describe Elasticsearch::Transport::Transport::Connections::Selector do
+
+  before do
+    class BackupStrategySelector
+      include Elasticsearch::Transport::Transport::Connections::Selector::Base
+
+      def select(options={})
+        connections.reject do |c|
+          c.host[:attributes] && c.host[:attributes][:backup]
+        end.sample
+      end
+    end
+  end
+
+  after do
+    Object.send(:remove_const, :BackupStrategySelector)
+  end
+
+  let(:backup_strategy_selector) do
+    BackupStrategySelector.new
+  end
+
+  describe 'the Random selector' do
+
+    let(:selector) do
+      described_class::Random.new(connections: [1, 2])
+    end
+
+    it 'is initialized with connections' do
+      expect(selector.connections).to eq([1, 2])
+    end
+
+    describe '#select' do
+
+      it 'returns a connection' do
+        expect(selector.select).to be_a(Integer)
+      end
+
+      context 'when multiple threads are used' do
+
+        it 'allows threads to select connections in parallel' do
+          threads = []
+          threads << Thread.new do
```

Comment:
50.times should be in outer loop to select connections in parallel

Reply:
Thanks for catching that. I missed updating this test when I updated the one below.
I've changed the test to match that one (https://github.com/elastic/elasticsearch-ruby/pull/665/files#diff-ea68875d75efe3a794afa867e510f560R102)